File: frontend/src/app/recipe/recipe.ts
import { map } from 'rxjs/operators';
import { Ingredient } from './ingredient';

export class Recipe {
  id: string;
  tags: string[];

  constructor(public ingredients: Ingredient[], public method: string[], public tagSet: Set<string>,
              public name: string, public image: any) {
    this.tags = Array.from(tagSet);
  }
}

File: backend/src/main/java/com/nina/mealplan/dm/Recipe.java
package com.nina.mealplan.dm;

import java.util.ArrayList;
import java.util.List;

import lombok.Data;

@Data
public class Recipe {

	private String id;
	private String name;
	private List<Ingredient> ingredients;
	private List<String> method;
	private List<String> tags = new ArrayList<String>();
	private String image;
}
